feature_engine/encoding/one_hot.py

```diff
@@ -174,6 +192,40 @@ def __init__(
                 f"Got {top_categories} instead"
             )
 
+        if top_categories is not None and custom_categories is not None:
+            raise ValueError(
+                "Both top_cagetories and custom_categories have values. "
+                "Only one of the two parameters may be used at a time. "
+                f"Got {top_categories} for top_categories. "
+                f"Got {custom_categories} for custom_categories."
+            )
+
+        if custom_categories and not isinstance(custom_categories, dict):
+            raise ValueError(
+                "custom_categories must be a dictionary. "
+                f"Got {custom_categories} instead."
+            )
+
+        if custom_categories:
+            # check that all values of custom_categories key-value pairs are lists
+            non_lists_custom_categories = [
+                val for val in custom_categories.values() if not isinstance(val, list)
+            ]
+            if len(non_lists_custom_categories) > 0:
+                raise ValueError(
+                    "custom_categories must be a dictionary that has lists as "
+                    f"its values. Got {custom_categories} instead."
+                )
+
+            # check that custom_categories variable match variables
```

Comment:
I'm reading this comment after responding to your prior question.

I agree that it will be a headache for us and useful for the user. I'll think about this more too.

Reply:
Alright then, let me know if you find an elegant solution :)